core/src/main/java/com/google/cloud/sql/core/DefaultConnectionInfoRepository.java

```diff
@@ -298,7 +299,14 @@ private InstanceMetadata fetchMetadata(CloudSqlInstanceName instanceName, AuthTy
 
         logger.debug(String.format("[%s] METADATA DONE", instanceName));
 
-        return new InstanceMetadata(ipAddrs, instanceCaCertificate);
+        return new InstanceMetadata(
+            instanceName,
+            ipAddrs,
+            Collections.singletonList(instanceCaCertificate),
+            "GOOGLE_MANAGED_CAS_CA".equals(instanceMetadata.getServerCaMode()),
+            instanceMetadata.getDnsName(),
+            instanceMetadata.getPscEnabled() != null
+                && instanceMetadata.getPscEnabled().booleanValue());
```

Comment:
Could we not use the fact that if the `ipAddrs` map contains an `IPType.PSC` entry then PSC is enabled...?

Reply:
Sure, but I want to keep instanceMetadata as a very simple value without any logic. So I'd rather compute pscEnabled here. 

I updated also updated logic on line 282 to only add the PSC address to the ipAddrs map if pscEnabled == true